src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -372,6 +395,11 @@ internal ImmutableArray<BoundExpression> ToImmutableAndFree()
             {
                 return init.ToImmutableAndFree();
             }
+
+            internal void Free()
```

Comment:
>Free [](start = 26, length = 4)

We should free `deconstructions`, `conversions` and `assignments` too.

Reply:
>We should free deconstructions, conversions and assignments too.

We always call `Consolidate` so the other three are always freed. Note that we have a fresh instance for every deconstruction.